### DISCUSSION TITLE
Greenfield API: Does it make sense that you need the `btcpay.store.canmodifystoresettings` permission to change an invoice's status?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2594] [General]
In order to use the API call "Mark invoice status" (i.e. set it to "Invalid" or "Complete") you need an API permission called 
 `btcpay.store.canmodifystoresettings`.

This sounds a bit strange as it's an invoice action and does not sound like it has something to do with store settings.

I ran into this while building the Zapier actions "Mark an Invoice as Invalid" and "Mark an Invoice as Complete".
Obviously I want to give Zapier as little permissions as possible, but now need to give full store settings access :( 

Maybe we need a new permission called `btcpay.store.canmodifyinvoice` ?

**Comment by Kukks** [comment]
We tend to try to reduce the number of permissions we create as it can cause a bit of confusion to end-users when creating keys. But I agree, this makes sense. ACK

**Reply by bolatovumar** [reply]
I have a draft PR [here](https://github.com/btcpayserver/btcpayserver/pull/2682/) which adds three new permissions for:

1. Updating invoice status
2. Archiving/unarchiving invoice
3. Updating invoice data

I added three permissions to keep them as specific as possible. If that's too much I can merge them all into one permission which will allow doing all of these things. Also, the existing "modify stores" permission will keep working as is for these actions so it's backwards compatible.

**Reply by Kukks** [reply]
I already have https://github.com/btcpayserver/btcpayserver/pull/2595

**Reply by bolatovumar** [reply]
Cool, looks like our PRs are very similar so I will just close min.
